samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoETSProviderImpl.java

```diff
@@ -71,20 +72,21 @@
     public      void                init                            () {
         LOG.info("init()");
 
-        String samigoFromAddress                    = serverConfigurationService.getString("samigo.fromAddress");
-
-        if( samigoFromAddress == null || StringUtils.isBlank(samigoFromAddress)){
-            fromAddress                             = serverConfigurationService.getString("setup.request", "no-reply@" + serverConfigurationService.getServerName());
-        } else {
-            fromAddress                             = samigoFromAddress;
+        fromAddress = serverConfigurationService.getString("samigo.fromAddress");
+        if(StringUtils.isBlank(fromAddress)){
+            String defaultAddress = "no-reply@" + serverConfigurationService.getServerName();
+            fromAddress = StringUtils.defaultIfBlank(serverConfigurationService.getString("setup.request"), defaultAddress);
         }
 
-        constantValues.put("localSakaiName" , serverConfigurationService.getString("ui.service", ""));
+        constantValues.put("localSakaiName" , StringUtils.defaultIfBlank(serverConfigurationService.getString("ui.service"), ""));
```

Comment:
As I found out the hard way during development, serverConfigurationService.getString(key, dflt) can actually still return null.